src/foam/core/Interface.js

```diff
@@ -107,21 +113,37 @@ foam.CLASS({
           ]
         });
 
-        if ( this.client )
-          foam.CLASS({
-            package: this.package,
-            name: 'Client' + this.name,
-            implements: [ this.id ],
-            flags: this.flags,
-            source: this.source,
-            properties: [
-              {
-                class: 'Stub',
-                of: this.id,
-                name: 'delegate'
-              }
-            ]
-          });
+      if ( this.client )
+        foam.CLASS({
+          package: this.package,
+          name: 'Client' + this.name,
+          implements: [ this.id ],
+          flags: this.flags,
+          source: this.source,
+          properties: [
+            {
+              class: 'Stub',
+              of: this.id,
+              name: 'delegate'
+            }
+          ]
+        });
+    
+      if ( this.nullStrategy )
+        foam.CLASS({
+          package: this.package,
+          name: 'Null' + this.name,
+          implements: [ this.id ],
+          flags: this.flags,
+          source: this.source,
+          axioms: [
+            {
+              class: 'foam.core.Null',
+              of: this.id,
+              name: 'implementor'
```

Comment:
This will prevent you have having more than one Null axiom because of the name collision. Can you make a factory for 'name' which is this.of + 'Implementor', then remove this line.